src/bootstrap/bin/main.rs

```diff
@@ -12,5 +12,26 @@ use bootstrap::{Build, Config};
 fn main() {
     let args = env::args().skip(1).collect::<Vec<_>>();
     let config = Config::parse(&args);
+    check_version(&config);
```

Comment:
Can we re-print this at the end of the build? I worry that in the not atypical case of quick screen-filling output the helpful message will otherwise get lost. And in cases where very little output is given, it seems not horrible to have a couple extra lines, maybe with some kind of prefix to make it clear it's a repeat.

Reply:
I changed this to print twice, but because rustbuild uses `std::process::exit`, it won't get printed if an error occurs.

Successful message:
```
$ x.py test tidy
warning: x.py has made several changes recently you may want to look at
help: consider looking at the changes in `src/bootstrap/CHANGELOG.md`
note: to silence this warning, add `changelog-seen = 1` to `config.toml`
Building stage0 tool tidy (x86_64-unknown-linux-gnu)
    Finished release [optimized + debuginfo] target(s) in 0.08s
tidy check
* 617 error codes
* highest error code: E0774
Checking which error codes lack tests...
Found 430 error codes
Found 0 error codes with no tests
Done!
* 298 features
fmt check
skip untracked path src/bootstrap/defaults/ during rustfmt invocations
skip untracked path src/tmp.txt during rustfmt invocations
skip untracked path tmp.rs during rustfmt invocations
skip untracked path x.rs during rustfmt invocations
skip untracked path y.rs during rustfmt invocations
warning: x.py has made several changes recently you may want to look at
help: consider looking at the changes in `src/bootstrap/CHANGELOG.md`
note: to silence this warning, add `changelog-seen = 1` to `config.toml`
note: this message was printed twice to make it more likely to be seen
Build completed successfully in 0:00:06
```

Message on error:
```
warning: x.py has made several changes recently you may want to look at
help: consider looking at the changes in `src/bootstrap/CHANGELOG.md`
note: to silence this warning, add `changelog-seen = 1` to `config.toml`
Building stage0 tool tidy (x86_64-unknown-linux-gnu)
    Finished release [optimized + debuginfo] target(s) in 0.09s
tidy check
* 617 error codes
* highest error code: E0774
Checking which error codes lack tests...
Found 430 error codes
Found 0 error codes with no tests
Done!
tidy error: /home/joshua/rustc/src/bootstrap/bin/main.rs:6: trailing whitespace
some tidy checks failed


command did not execute successfully: "/home/joshua/rustc/build/x86_64-unknown-linux-gnu/stage0-tools-bin/tidy" "/home/joshua/rustc" "/home/joshua/rustc/build/x86_64-unknown-linux-gnu/stage0/bin/cargo"
expected success, got: exit code: 1


failed to run: /home/joshua/rustc/build/bootstrap/debug/bootstrap test tidy
Build completed unsuccessfully in 0:00:03
```

Maybe it could be moved to `bootstrap.py` or `x.py` instead?

Reply:
Hm let's avoid that for now.